flake.nix

```diff
@@ -1,30 +1,58 @@
 {
   description = "Seabug";
+
   inputs = {
-    nixpkgs = { url = "github:NixOS/nixpkgs"; };
-    cardano-node.url = "github:input-output-hk/cardano-node/1.35.1";
-    flake-utils = { url = "github:numtide/flake-utils"; };
-    # https://github.com/hercules-ci/arion/pull/153
-    arion = {
-      url = "github:hercules-ci/arion/bd3e2fe4e372d0b5f965f25f27c5eb7c1a618c4a";
-    };
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-unstable;
+
+    # TODO change after they are merged
+    nft-marketplace-frontend.url = github:mlabs-haskell/nft-marketplace/aciceri/contract-updates-and-nix;
+
+    nft-marketplace-server.url = github:mlabs-haskell/nft-marketplace-server/aciceri/nix;
   };
-  outputs = { self, nixpkgs, cardano-node, flake-utils, arion }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let pkgs = nixpkgs.legacyPackages.${system};
-      in rec {
-        devShell = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; [
-            arion.packages.${system}.arion
-            cardano-node.packages.${system}.cardano-cli
-            curl
-            httpie
-            ipfs
-            jq
-            postgresql
-            shfmt
-            expect
+  outputs =
+    { self
+    , nixpkgs
+    , nft-marketplace-frontend
+    , nft-marketplace-server
+    } @ inputs:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+
+      pkgsFor = lib.genAttrs supportedSystems (system: nixpkgs.legacyPackages.${system});
+
+      lib = nixpkgs.lib.extend (self: super: {
+        perSystem = super.genAttrs supportedSystems;
```

Comment:
this is too convoluted, just put it in the `let` above

Reply:
convoluted in terms of readability or evaluation efficiency?